### ISSUE TITLE
Intermittent crash 

### ISSUE THREAD

**Issue by er1czz** [wkentaro/labelme#1194]
### Provide environment information

/Library/miniconda3/bin/python
Python 3.8.12

### What OS are you using?

macOS Monterey Version 12.5

### Describe the Bug

During label editing, labelme would crash unexpectedly.

I tested both version 4.5.12 and 5.0.1
4.5.12 is the latest version of labelme that can be installed through brew.
5.0.1 is installed as a stand-alone app, directly downloaded from [the release](https://github.com/wkentaro/labelme/releases)

It used to happen once or twice per day when I was working on polygon or box labels.
I start to work on point labels recently, and the crash happened much more frequently, like every 15 minuets. 

Error message in command line:

Traceback (most recent call last):
  File "app.py", line 1885, in removeSelectedPoint
AttributeError: 'NoneType' object has no attribute 'points'
zsh: abort      labelme

### To Reproduce

The crash occurs intermittently while I was creating / editing point labels.

**Comment by wanghao14** [comment]
I have encountered the same problem. 
My environment information is: Python 3.8.12 and macOS 11.6. I installed labelme by `pip install labelme` with version 5.0.1 installed.

**Comment by wanghao14** [comment]
@wkentaro This Indexerror is more likely to occur when doing deleting point with dense points, which greatly affects my data processing, and I hope to get your attention.
<img width="990" alt="image" src="https://user-images.githubusercontent.com/31157833/201508455-78ab7ff4-c928-43c5-829f-e7c996dca357.png">

**Comment by wkentaro** [comment]
Yeah, but it seems hard to reproduce, and I don't think I have time to reproduce and fix.

**Comment by wanghao14** [comment]
@wkentaro Thanks for your reply! I've uploaded a video recording the issue when it happened and the images and json files involved at https://drive.google.com/drive/folders/1FNJhtm-kyZgR_Bow5KX0OkhkkIZuDE1B, please check it if it's convenient for you. Any ideas on possible solutions would be greatly appreciated.

**Comment by er1czz** [comment]
@wkentaro 
Thank you so much for looking into this issue!
My workaround solution is to delete the whole label (from the label panel) instead of single points on the image.
This approach is relatively straightforward for point labels. 
However, it is probably impractical for polygon masks, such as the one shown in @wanghao14 's video.

**Comment by wkentaro** [comment]
> *This was generated by AI during triage.*

Moving to **needs-triage** (appears resolved). The crash was `AttributeError: 'NoneType' object has no attribute 'points'` in the old `removeSelectedPoint` while deleting points. The refactored remove-point flow now guards against that: `Canvas.remove_selected_point` returns early `if shape is None or index is None`, and `MainWindow.remove_selected_point` only touches `hovered_shape.points` behind a truthy `hovered_shape` check (`if canvas.hovered_shape and len(canvas.hovered_shape.points) == 0`). So deleting points no longer dereferences a `None` shape. (Related hardening also went in for #968.) Recommend closing unless deleting points still aborts on current `main`.
